Session.js

```diff
@@ -682,11 +682,25 @@ Session.prototype = {
 		if (typeof args !== 'undefined' && !Array.isArray(args)) {
 			throw new Error('Arguments passed to execute must be an array');
 		}
+		var self = this;
 
-		var result = this._post('execute', {
-			script: util.toExecuteString(script),
-			args: args || []
-		}).then(lang.partial(convertToElements, this), fixExecuteError);
+		var result = this.capabilities.useExecuteSyncEndpoint
+			?  executeWithEndpoint('execute/sync')
+			: executeWithEndpoint('execute');
+
+		function executeWithEndpoint(endpoint) {
+			return self._post(endpoint, {
+				script: util.toExecuteString(script),
+				args: args || []
+			}).then(lang.partial(convertToElements, self), fixExecuteError).catch(function(error) {
+				if (error.detail.error === 'unknown command'
+					&& endpoint !== 'execute/sync') {
```

Comment:
For clarity, check whether the capability (`useExecuteSyncEndpoint`) has been set rather than checking the endpoint value.

Reply:
Duh. That's been the pattern the whole time. Fixed. I am going to run some tests to make sure I didn't break anything.